server/src/main/java/org/opensearch/indices/replication/checkpoint/PublishCheckpointAction.java

```diff
@@ -199,6 +199,9 @@ protected void shardOperationOnReplica(PublishCheckpointRequest request, IndexSh
         Objects.requireNonNull(replica);
         ActionListener.completeWith(listener, () -> {
             logger.trace(() -> new ParameterizedMessage("Checkpoint {} received on replica {}", request, replica.shardId()));
+            if (replica.indexSettings().isRemoteNode() == false) {
+                logger.trace("Received segrep checkpoint on a docrep shard copy during an ongoing remote migration. NoOp.");
+            }
```

Comment:
this means that a doc rep node is receiving the segrep checkpoint, can this happen for any reason?

Reply:
We need to add ` return new ReplicaResult();` here ?

Reply:
Done in latest rev